src/isolate.rs

```diff
@@ -345,6 +361,40 @@ impl Isolate {
     }
   }
 
+  /// Add a callback to invoke in case the heap size is close to the heap limit.
+  /// If multiple callbacks are added, only the most recently added callback is
+  /// invoked.
+  ///
+  /// # Safety
+  ///
+  /// Don't pass a null-pointer for the callback!
+  /// Make sure the callback only casts `data` to the type the `data` argument
+  /// here.
+  pub unsafe fn add_near_heap_limit_callback(
```

Comment:
I don't think this needs to be unsafe.

Reply:
yeah, the caller might have to do some casting but i'm thinking of making a generics-based safe impl in Deno for this

Reply:
Generics based safe is a bit tricky. You'd have to pass an owned pointer like `Box<T>` or `Rc<T>`. This thing would need to be dropped when the isolate is disposed or the heap limit callback is removed.

Reply:
that's the idea